src/Files.Uwp/ViewModels/FolderSettingsViewModel.cs

```diff
@@ -436,7 +436,7 @@ private static LayoutPreferences ReadLayoutPreferencesFromSettings(string folder
                 ApplicationDataCompositeValue adcv = (ApplicationDataCompositeValue)dataContainer.Values[folderPath];
                 return LayoutPreferences.FromCompositeValue(adcv);
             }
-            else if (folderPath == CommonPaths.DownloadsPath)
+            else if (folderPath.Replace('_', '\\') == CommonPaths.DownloadsPath)
```

Comment:
Ok lol this "works" but will fail if the original path does contain a "_", also should be done on line 453. My suggestion would be to spit this method into two. One that reads from settings and returns null if layout is not found [426-438] and a second that returns the default layout give the path [438-470].

Reply:
Two separate methods? so I can do if instead of else if on line 439. But before that line, since I already used the folderPath with '_', I would already replace and after line 439 down it would be '\\'.

Reply:
Sorry something like (writing on the fly/untested):
```cs
public static LayoutPreferences GetLayoutPreferencesForPath(string folderPath)
        {
            IUserSettingsService userSettingsService = Ioc.Default.GetService<IUserSettingsService>();
            if (userSettingsService.PreferencesSettingsService.AreLayoutPreferencesPerFolder)
            {
                var layoutPrefs = ReadLayoutPreferencesFromAds(folderPath.TrimEnd('\\'));
                return layoutPrefs ?? 
                    ReadLayoutPreferencesFromSettings(folderPath.TrimEnd('\\').Replace('\\', '_')) ??
                    GetDefaultPreferences(folderPath.TrimEnd('\\')) ; <---- new method
            }

            return LayoutPreferences.DefaultLayoutPreferences;
        }

private static LayoutPreferences ReadLayoutPreferencesFromSettings(string folderPath)
        {
            if (string.IsNullOrEmpty(folderPath))
            {
                return LayoutPreferences.DefaultLayoutPreferences;
            }

            IUserSettingsService userSettingsService = Ioc.Default.GetService<IUserSettingsService>();
            ApplicationDataContainer dataContainer = localSettings.CreateContainer("LayoutModeContainer", ApplicationDataCreateDisposition.Always);
            folderPath = new string(folderPath.TakeLast(254).ToArray());
            if (dataContainer.Values.ContainsKey(folderPath))
            {
                ApplicationDataCompositeValue adcv = (ApplicationDataCompositeValue)dataContainer.Values[folderPath];
                return LayoutPreferences.FromCompositeValue(adcv);
            }

            return null;
}

private static LayoutPreferences GetDefaultPreferences(string folderPath) <--
{
            if (folderPath) == CommonPaths.DownloadsPath)
            {
                // Default for downloads folder is to group by date created
                return new LayoutPreferences
                {
                    ...
                };
            }
            else if (LibraryHelper.IsLibraryPath(folderPath))
            {
                // Default for libraries is to group by folder path
                return new LayoutPreferences
                {
                    .....
                };
            }
            else
            {
                return LayoutPreferences.DefaultLayoutPreferences; // Either global setting or smart guess
            }
        }
```

Reply:
It's done in the last commit